pkg/lifecycle/render/helm/commands.go

```diff
@@ -37,16 +46,45 @@ func (h *helmCommands) Init() error {
 	return errors.Wrapf(err, "helm init: %s", output)
 }
 
-func (h *helmCommands) DependencyUpdate(chartRoot string) error {
-	dependencyArgs := []string{
-		"update",
-		chartRoot,
+func (h *helmCommands) MaybeDependencyUpdate(chartRoot string) error {
+	debug := level.Debug(log.With(h.logger, "method", "maybeDependencyUpdate"))
+
+	requirementsExists, err := h.fs.Exists(filepath.Join(chartRoot, "requirements.yaml"))
+	if err != nil {
+		return errors.Wrap(err, "check requirements yaml existence")
 	}
-	dependencyCommand, err := helm.NewDependencyCmd(dependencyArgs)
+
+	if !requirementsExists {
+		return nil
+	}
+
+	requirementsB, err := h.fs.ReadFile(filepath.Join(chartRoot, "requirements.yaml"))
 	if err != nil {
-		return err
+		return errors.Wrap(err, "read requirements yaml")
 	}
-	return dependencyCommand.Execute()
+
+	requirements := chartutil.Requirements{}
+	if err := yaml.Unmarshal(requirementsB, &requirements); err != nil {
+		return errors.Wrap(err, "unmarshal requirements yaml")
+	}
+
+	allEmpty := true
```

Comment:
the only case i really want to deal with right now is where all the repository fields are empty - e.g. istio